icicle_v3/tests/test_curve_api.cpp

```diff
@@ -86,18 +91,34 @@ class CurveApiTest : public ::testing::Test
       oss << dev_type << " " << msg;
 
       auto config = default_msm_config();
+      config.c = c;
       config.batch_size = batch;
+      config.are_bases_shared = true;
+      config.precompute_factor = precompute_factor;
+
+      ConfigExtension ext;
+      ext.set("n_threads", n_threads);
+      config.ext = &ext;
+
+      // Note: allocating the precompute_bases on device since CUDA backend assumes that.
+      // TODO: fix CUDA backend to support host memory too.
+      A* precomp_bases = nullptr;
+      ICICLE_CHECK(icicle_malloc((void**)&precomp_bases, N * precompute_factor * sizeof(A)));
+      ICICLE_CHECK(msm_precompute_bases(bases.get(), N, config, precomp_bases));
+
       START_TIMER(MSM_sync)
       for (int i = 0; i < iters; ++i) {
-        msm(scalars.get(), bases.get(), N, config, result);
+        ICICLE_CHECK(msm(scalars.get(), precomp_bases, N, config, result));
```

Comment:
need to set config.bases_on_device=true since icicle_malloc() allocates on the device.

Reply:
Added the proper config